src/containers/Firmware/index.js

```diff
@@ -0,0 +1,419 @@
+      key: 'editAssignment',
+      width: 60,
+      render: (_, record) => (
+        <Button
+          title={`edit-track-${record.modelId}`}
+          className={styles.InfoButton}
+          type="primary"
+          icon={<FormOutlined />}
+          onClick={() => {
+            setTaskAssignmentValues({ ...record });
+            setEditAssignmentModal(true);
+          }}
+        />
+      ),
+    },
+    {
+      title: '',
+      dataIndex: '',
+      key: 'deleteAssignment',
+      width: 60,
+      render: (_, record) => (
+        <Button
+          title={`delete-track-${record.modelId}`}
+          className={styles.InfoButton}
+          type="primary"
+          icon={<DeleteFilled />}
+          onClick={() => {
+            setTaskAssignmentValues({ ...record });
+            setDeleteAssignmentModal(true);
+          }}
+        />
+      ),
+    },
+  ];
+
+  const versionColumn = [
+    {
+      title: 'MODEL ID',
+      dataIndex: 'modelId',
+      key: 'modelIdFirmware',
+      width: 100,
+    },
+    {
+      title: 'VERSION',
+      dataIndex: 'versionName',
+      key: 'versionFirmware',
+      width: 225,
+    },
+    {
+      title: 'DESCRIPTION',
+      dataIndex: 'description',
+      key: 'description',
+      width: 350,
+    },
+    {
+      title: 'COMMIT',
+      dataIndex: 'commit',
+      key: 'commit',
+      width: 100,
+    },
+    {
+      title: 'RELEASE DATE',
+      dataIndex: 'releaseDate',
+      key: 'date',
+      width: 180,
+      render: time => moment(time, 'x').format('DD MMM YYYY, hh:mm a'),
+    },
+    {
+      title: '',
+      dataIndex: '',
+      key: 'editFirmware',
+      width: 60,
+      render: (_, record) => (
+        <Button
+          title={`edit-firmware-${record.modelId}`}
+          className={styles.InfoButton}
+          type="primary"
+          icon={<FormOutlined />}
+          onClick={() => {
+            setFirmwareValues({ ...record });
+            setEditVersionModal(true);
+          }}
+        />
+      ),
+    },
+    {
+      title: '',
+      dataIndex: '',
+      key: 'deleteFirmware',
+      width: 60,
+      render: (_, record) => {
+        const found = Object.values(trackAssignmentData).some(
+          i => Object.values(record).indexOf(i.modelId) > 0
+        );
+        return !found ? (
+          <Button
+            title={`delete-firmware-${record.modelId}`}
+            className={styles.InfoButton}
+            type="primary"
+            icon={<DeleteFilled />}
+            onClick={() => {
+              setFirmwareValues({ ...record });
+              setDeleteVersionModal(true);
+            }}
+          />
+        ) : null;
+      },
+    },
+  ];
+
+  const trackAssignmentReady =
+    !trackAssignmentLoading &&
+    !firmwareTrackLoading &&
+    !firmwareModelLoading &&
+    Object.keys(trackAssignmentError).length === 0 &&
+    Object.keys(firmwareTrackError).length === 0 &&
+    Object.keys(firmwareModelError).length === 0;
```

Comment:
change to `const trackAssignmentReady =
    !trackAssignmentLoading && ! trackAssignmentError `

the others are related to the Modals not the table